src/main/kotlin/org/rust/lang/core/type/RustResolvedType.kt

```diff
@@ -0,0 +1,26 @@
+package org.rust.lang.core.type
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.SyntheticElement
+import org.rust.lang.core.psi.RustImplItem
+import org.rust.lang.core.psi.RustImplMethodMember
+import org.rust.lang.core.psi.impl.mixin.isStatic
+
+
+interface RustResolvedType: PsiElement, SyntheticElement {
```

Comment:
An excellent question! There it is not needed in this at all, fixed :) 

Looks like I originally implemented this because I was doing PSI_MODIFICATION_COUNT based caching "inside" the types themselves. But now caching happens on the real PSI elements, which is the right way I think.